File: 프로그래머스/python/LV1/약수의 개수와 덧셈.py
def solution(left, right):
    answer = 0
    for i in range(left, right + 1):
        num = count(i)
        if num % 2 == 0:
            answer += i
        else:
            answer -= i
    return answer


def count(n):
    ans = 0
    for i in range(1, n + 1):
        if n % i == 0:
            ans += 1
    return ans

File: 프로그래머스/javascript/LV1/약수의 개수와 덧셈.js
function counter(number) {
  let answer = 0;
  for (let i = 1; i <= number; i++) {
    if (number % i === 0) {
      answer += 1;
    }
  }
  return answer;
}

function solution(left, right) {
  let total = 0;
  for (let i = left; i <= right; i++) {
    if (counter(i) % 2 === 0) {
      total += i;
    } else {
      total -= i;
    }
  }
  return total;
}
